src/components/Button.jsx

```diff
@@ -0,0 +1,47 @@
+import  React, {useState, useEffect} from 'react'
+import PropTypes from 'prop-types'
+import '../App.css'
+
+const  Button = (props) => {
+    const [wide, setWide] = useState('100px');
+    useEffect(()=>{
+        if(props.wide) setWide('200px');
+    });
+  return  <button class='button' id={props.id}
+        style={ { padding: '1em',
+        border: 'none',
+        width: wide,
+        fontSize: '.8em',
+        margin: '.1em',
+        backgroundColor: props.color,
+        borderRadius: '2em',
+        cursor: 'pointer'
```

Comment:
[optional]
This could get ugly really quick. if you want to write styles in the component, look at styled component `emotion styled component` is an example of one.